### PR TITLE
Fixes #4455 Implement buttons for Mattermost channel

### LINKED ISSUE #4455: Implement buttons for Mattermost channel
**Description of Problem**:

Buttons in Mattermost appear currently only as text message, not as clickable buttons. See also 
this thread in the [forum](https://forum.rasa.com/t/rasa-buttons-not-appearing-in-mattermost/12984/3).

**Overview of the Solution**:

Changes are necessary in 
* rasa\core\channels\mattermost.py
    * Implement `send_text_with_buttons()` in `MattermostBot `class according to [mattermost message buttons docs](https://docs.mattermost.com/developer/interactive-messages.html#message-buttons)
   * Initialize `MattermostBot `and `MattermostInput `with an additional parameter `webhook_url`
* rasa\cli\initial_project\credentials.yml (not really necessary, but very helpful for an easy start)
    * add Mattermost example with additional `webhook_url`

**Examples** (if relevant):

Current state:
image

Future state:
image

**Blockers** (if relevant):

**Definition of Done**:

- [ ] Tests are added
- [ ] Feature described the doc
- [ ] Feature mentioned in the changelog

### PR DESCRIPTION
**Pull request by mexophantes**
**Proposed changes**:
   - rasa\core\channels\mattermost.py
      - Implement send_text_with_buttons() in MattermostBot class according to mattermost message buttons docs
      - Initialize MattermostBot and MattermostInput with an additional parameter webhook_url
   - rasa\cli\initial_project\credentials.yml (not really necessary, but very helpful for an easy start)
      - add Mattermost example with additional webhook_url

**Status (please check what you already did)**:
- [x] made PR ready for code review
- [x] added some tests for the functionality
- [x] updated the documentation
- [x] updated the changelog
- [x] reformat files using `black` (please check [Readme](https://github.com/RasaHQ/rasa_nlu#code-style) for instructions)

### CODE REVIEW

**Comment by mexophantes** [comment]
@tabergma Resolved your comments from PR #4456 here.

File: rasa/core/channels/mattermost.py [hunk 1]
```diff
@@ -86,16 +156,27 @@ def blueprint(self, on_new_message):
...
-            if output:
+
+            if not output:
+                return response.text("")
+
+            sender_id = output["user_id"]
+            self.bot_channel = output["channel_id"]
+                
+            # handle normal message with trigger_word
+            if "trigger_word" in output:
```
Comment (tabergma) [inline rasa/core/channels/mattermost.py]: To reduce the complexity of the method, you could put the code of the if block into a separate method as well as the code from the else block.

Comment (mexophantes) [reply rasa/core/channels/mattermost.py]: Ok, done as well. 
I additionally added the possibility to send images as attachments.

Code is formatted with black.

File: rasa/core/channels/mattermost.py [hunk 2]
```diff
@@ -86,27 +216,21 @@ def blueprint(self, on_new_message):
...
-                        "Exception when trying to handle message.{0}".format(e)
-                    )
-                    logger.debug(e, exc_info=True)
-                    pass
-            return response.text("")
+
+            if not output:
+                return response.text("")
+
+            sender_id = output["user_id"]
```
Comment (tabergma) [inline rasa/core/channels/mattermost.py]:
```suggestion
            
```
Can be removed as it is never used. Reason for travis failure.

Comment (mexophantes) [reply rasa/core/channels/mattermost.py]: True, leftovers from the refactoring...

File: rasa/core/channels/mattermost.py [hunk 3]
```diff
@@ -70,13 +136,77 @@ def __init__(self, url: Text, team: Text, user: Text, pw: Text) -> None:
...
+                self.webhook_url,
+            )
+            user_msg = UserMessage(
+                text, out_channel, sender_id, input_channel=self.name()
+            )
+            await on_new_message(user_msg)
+        except Exception as e:
+            logger.error("Exception when trying to handle message.{0}".format(e))
+            logger.debug(e, exc_info=True)
+            pass
```
Comment (tabergma) [inline rasa/core/channels/mattermost.py]: Can be removed.

File: rasa/core/channels/mattermost.py [hunk 4]
```diff
@@ -39,6 +55,53 @@ def __init__(self, url, team, user, pw, bot_channel):
         json_message.setdefault("message", "")
         self.post("/posts", json_message)
 
+    async def send_image_url(
+        self, recipient_id: Text, image: Text, **kwargs: Any
+    ) -> None:
+        """Sends an image. """
```
Comment (tabergma) [inline rasa/core/channels/mattermost.py]:
```suggestion
        """Sends an image."""
```

File: rasa/core/channels/mattermost.py (hunk 3 above)
Comment (mexophantes) [reply rasa/core/channels/mattermost.py]: Are you sure? Where will the exception be handled if I remove this?

**Comment by mexophantes** [comment]
> Thanks for resolving all my comments. Just one more thing: Can you please add a changelog entry? And also update the documentation here https://github.com/RasaHQ/rasa/blob/master/docs/user-guide/connectors/mattermost.rst. Once that is done we can merge the PR 🚀

Done

File: rasa/core/channels/mattermost.py (hunk 3 above)
Comment (tabergma) [reply rasa/core/channels/mattermost.py]: I meant, you can remove the `pass`, not the exception :)

Comment (mexophantes) [reply rasa/core/channels/mattermost.py]: Alright. That's true. Removed it in both places.

**Pull request merged**